packages/ui/src/machines/authenticator/actors/signIn.ts

```diff
@@ -577,7 +577,8 @@ export function signInActor({ services }: SignInMachineOptions) {
           const { user } = context;
           const { confirmation_code } = context.formValues;
 
-          return Auth.verifyTotpToken(user, confirmation_code);
+          await Auth.verifyTotpToken(user, confirmation_code);
+          return await Auth.currentAuthenticatedUser();
```

Comment:
The return type is Promise<any>. And that's because we are inferring the return type from `Auth.currentAuthenticatedUser`, which apparently doesn't have one. 

<img width="849" alt="image" src="https://user-images.githubusercontent.com/65630/178832683-dcdaeb5b-6a1a-4b04-9d02-55662714ea09.png">

The results should be `Promise<any>` just as we do with `confirmSignIn` which has the same pattern of handling the sign in and then calling `Auth.currentAuthenticatedUser`. 

We then take the results and assign it to `setUser` action here after the verifyTOTP finishes..

```
 invoke: {
                src: 'verifyTotpToken',
                onDone: {
                  actions: [
                    'setUser',
                    'clearChallengeName',
                    'clearRequiredAttributes',
                  ],
                  target: '#signInActor.resolved',
                },
```

I think we keep the naming, since it's the same pattern we did with `confirmSignIn` action.

```
async confirmSignIn(context, event) {
          const { challengeName, user } = context;
          const { confirmation_code: code } = context.formValues;

          let mfaType;
          if (
            challengeName === AuthChallengeNames.SMS_MFA ||
            challengeName === AuthChallengeNames.SOFTWARE_TOKEN_MFA
          ) {
            mfaType = challengeName;
          }

          await services.handleConfirmSignIn({ user, code, mfaType });
          return await Auth.currentAuthenticatedUser();
        },
```

Reply:
OK, here is the change so we don't break existing users, and now the object will have all the fields for both. 2461f5c6b7c181ba14f516649653e58a15a787cb

Reply:
Updated code to return `Auth.verifyTotpToken` however we no longer `setUser` on this.